src/Subscriptions/resources/hooks/useDonationsBySubscription.ts

```diff
@@ -0,0 +1,26 @@
+import { useState, useEffect } from 'react';
+import { useEntityRecords } from '@wordpress/core-data';
+import apiFetch from '@wordpress/api-fetch';
+import { Donation } from '@givewp/donations/admin/components/types';
+
+/**
+ * @unreleased
+ */
+export function useDonationsBySubscription(
+    subscriptionId: number,
+    mode: 'test' | 'live',
+) {
+    const queryArgs = {
+        subscriptionId,
+        mode,
+    };
+
+    const entityResult = useEntityRecords('givewp', 'donation', queryArgs);
+
+    return {
+        records: entityResult.records as Donation[] | null,
+        record: entityResult.records?.[0] as Donation | null,
```

Comment:
is this `record` key being used? For getting a single record we should be passing an id to `useEntityRecord` as this one would just be grabbing the first record which I don't think is necessary to abstract

Reply:
No unnecessary for this PR.
Resolved: 253ac3e1c5351763c283449bec017dd97b6404d2